src/Orleans/Providers/IProviderRuntime.cs

```diff
@@ -89,4 +92,48 @@ public interface ILogConsistencyProviderRuntime : IProviderRuntime
     /// <returns>The result of invocation, which will be returned to the client.</returns>
     public delegate Task<object> InvokeInterceptor(
         MethodInfo targetMethod, InvokeMethodRequest request, IGrain target, IGrainMethodInvoker invoker);
+
+    /// <summary>
+    /// The delegate invoked to execute the request.
+    /// </summary>
+    /// <returns>A <see cref="Task"/> which must be awaited before processing continues.</returns>
+    public delegate Task MethodInvocationDelegate();
```

Comment:
Just a thought, but should we call them Filters (or GrainCallFilters or something like that) now that it fully aligns with action filters?